### ISSUE TITLE
[Question/Help Wanted] Non-deterministic training results in `StackCube-v1` with SAC

### ISSUE THREAD

**Issue by songyuc** [haosulab/ManiSkill#1250]
Hello ManiSkill team and contributors,

First, thank you for developing and maintaining such an excellent and powerful benchmark! It's been an invaluable resource for my research.

I am encountering a problem regarding **deterministic reproducibility** when training a SAC agent on the `StackCube-v1` task. I cannot achieve identical results between two separate training runs launched with the same command.

### 🐛 Describe the bug
The final evaluation metric `success_once` varies significantly between identical runs. This makes it difficult to reliably compare algorithms or hyperparameters.

### 📜 Reproduction Steps & Code
**My exact training command:**
    ```bash
    python sac.py --env_id="StackCube-v1" \
      --num_envs=32 --utd=0.5 --buffer_size=500_000 \
      --total_timesteps=500_000 --eval_freq=50_000 \
      --control-mode="pd_ee_delta_pos"
    ```

#### 📈 Observed Results
I conducted two independent training runs with the same setup. Here is a comparison of the `success_once` metric at different stages:

**Run 1:**
```
success_once: 0.00, return: 2.59: 0%
...
success_once: 0.88, return: 39.84: 90%
success_once: 0.88, return: 39.84: : 500032lt [1:04:37, 132.17lt/s]
```

**Run 2:**
```
success_once: 0.00, return: 2.59: 0%
...
success_once: 0.31, return: 36.46: 90%
success_once: 0.31, return: 36.46: : 500032lt [43:38, 199.09lt/s]
```
The final success rate differs drastically (0.88 vs. 0.31).

#### 💡 Expected behavior
I expect that after setting all random seeds, two identical training runs should produce numerically identical results (or very nearly so) for all metrics, including `success_once`.

### 📋 Environment information
*   **OS:** Ubuntu 22.04 LTS
*   **Python version:** Python 3.11.13
*   **PyTorch version:** 2.7.1+cu118
*   **CUDA/cuDNN version:** 12.8/9.1
*   **ManiSkill version:** 3.0.0b21
*   **Other relevant libs:** gym==0.29.1, numpy==1.26.4

### ❓ Question
Could you please provide guidance on the **correct and complete procedure to ensure full training reproducibility** for a given seed in ManiSkill?
Specifically:
1.  What is the recommended way to seed the **vectorized environment** (`num_envs > 1`) to ensure each parallel environment is initialized deterministically?
2.  Are there any other potential sources of non-determinism within ManiSkill or its dependencies (e.g., SAPIEN) that I need to account for?
3.  Is achieving such a high level of determinism (identical results across runs) expected and supported?

Any advice, code snippets, or pointers to relevant documentation would be immensely helpful. Thank you for your time and support!

Best regards,
Yucheng Song

**Comment by StoneT2000** [comment]
Found the issue, https://github.com/haosulab/ManiSkill/tree/main/examples/baselines/sac/sac.py#L439 is actually using the numpy action sampler...
